src/extension/prompt/node/chatParticipantRequestHandler.ts

```diff
@@ -260,11 +285,13 @@ export class ChatParticipantRequestHandler {
 	}
 
 	private async selectIntent(command: CommandDetails | undefined, history: Turn[]): Promise<IIntent> {
-		if (!command?.intent && this.location === ChatLocation.Editor) { // TODO@jrieken do away with location specific code
+		if (!command?.intent && (this.location === ChatLocation.Editor || this.location === ChatLocation.Notebook)) { // TODO@jrieken do away with location specific code
 
 			let preferredIntent: Intent | undefined;
-			if (this.location === ChatLocation.Editor && this.documentContext && this.request.attempt === 0 && history.length === 0) { // TODO@jrieken do away with location specific code
-				if (this.documentContext.selection.isEmpty && this.documentContext.document.lineAt(this.documentContext.selection.start.line).text.trim() === '') {
+			if (this.documentContext && this.request.attempt === 0 && history.length === 0) {
```

Comment:
the intent is only selected this way for the first request (though I don't really understand why), so making a follow up request in the same inline chat won't have the correct prompt.